### PR TITLE
fix(client): stack trace with source-map-support. Closes #3945

### LINKED ISSUE #3945: Error in prisma results in unhandled promise rejection
When there's an invalid value provided to prisma at runtime, it will result in an error, but that error doesn't contain a useful stack trace.

```typescript
import "source-map-support/register"
import { prisma } from "./prisma/prisma"

prisma.$connect().catch((err) => {
	console.log(err)
	process.exit(1)
}).then(() => {
	return prisma.user.findOne({ where: { id: null as any as string }})
}).then(() => {
	process.exit(0)
})
```

```
❯ ts-node app.ts
(node:46264) UnhandledPromiseRejectionWarning: Error:
Invalid `prisma.user.findOne()` invocation in
/Users/prisma/projects/redacted/backend/app.ts:10:33

6    // process.exit(1)
7 }).then(() => {
8    return prisma.user.findOne({ where: { id: null as any as string }})
→ 9 }).then(() => {
where: {
id: null
~~~~
}
})

Argument id for where.id must not be null. Please use undefined instead.


    at Qh.validate (/Users/prisma/projects/redacted/backend/node_modules/@prisma/client/runtime/src/runtime/query.ts:280:19)
    at t._executeRequest (/Users/prisma/projects/redacted/backend/node_modules/@prisma/client/runtime/src/runtime/getPrismaClient.ts:883:16)
    at /Users/prisma/projects/redacted/backend/node_modules/@prisma/client/runtime/src/runtime/getPrismaClient.ts:787:16
    at AsyncResource.runInAsyncScope (async_hooks.js:186:9)
    at t._request (/Users/prisma/projects/redacted/backend/node_modules/@prisma/client/runtime/src/runtime/getPrismaClient.ts:786:25)
    at Object.then (/Users/prisma/projects/redacted/backend/node_modules/@prisma/client/runtime/src/runtime/getPrismaClient.ts:948:39)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:46264) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:46264) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Since this is an unhandled promise rejection, no error is returned and you can't catch it, which is probably the same reason for not getting a useful stacktrace about where the error originated from.

```
Environment variables loaded from prisma/.env
@prisma/cli          : 2.9.0
@prisma/client       : 2.9.0
Current platform     : darwin
Query Engine         : query-engine 369b3694b7edb869fad14827a33ad3f3f49bbc20 (at node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 369b3694b7edb869fad14827a33ad3f3f49bbc20 (at node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 369b3694b7edb869fad14827a33ad3f3f49bbc20 (at node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 369b3694b7edb869fad14827a33ad3f3f49bbc20 (at node_modules/@prisma/cli/prisma-fmt-darwin)
Studio               : 0.296.0
Preview Features     : atomicNumberOperations
✨  Done in 2.53s.
```

### PR DESCRIPTION
**Pull request by timsuchanek**
Closes #3945

The parsing logic of the stack trace is now moved out to a separate function called `parseStack`.
Also we're printing the "simple" stack without the concrete stack trace, if the line, that we found doesn't contain Prisma.

### CODE REVIEW

File: src/docker/docker-compose.yml [hunk 1]
```diff
@@ -48,10 +48,13 @@ services:
       - MYSQL_DATABASE=tests
       - MYSQL_USER=prisma
       - MYSQL_PASSWORD=prisma
+    # https://stackoverflow.com/questions/55559386/how-to-fix-mbind-operation-not-permitted-in-mysql-error-log
+    cap_add:
+      - SYS_NICE # CAP_SYS_NICE
```
Comment (Jolg42) [inline src/docker/docker-compose.yml:53]: Looks unrelated, first time I see that.
Did it happen on your Linux machine?

File: src/packages/client/package.json [hunk 2]
```diff
@@ -113,6 +113,7 @@
     "rollup": "2.32.1",
     "rollup-plugin-dts": "1.4.13",
     "set-value": "3.0.2",
+    "source-map-support": "^0.5.19",
```
Comment (Jolg42) [inline src/packages/client/package.json:116]:
```suggestion
    "source-map-support": "0.5.19",
```
Because renovate will create a PR to do that if not 😄

File: src/packages/client/src/__tests__/integration/errors/source-map-support/test.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import 'source-map-support/register'
+import { getTestClient } from '../../../../utils/getTestClient'
+
+// X is here to have a different stack trace in js and ts
+type X = {}
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/integration/errors/source-map-support/test.ts:5]: I think can only see a snapshot for TS though. Is it interesting to have one for JS too?

**Review by Jolg42 — APPROVED** [review]
See comments, looks good!

File: src/docker/docker-compose.yml (hunk 1 above)
Comment (timsuchanek) [reply src/docker/docker-compose.yml:53]: Yes, happened on my Linux machine

File: src/packages/client/src/__tests__/integration/errors/source-map-support/test.ts (hunk 3 above)
Comment (timsuchanek) [reply src/packages/client/src/__tests__/integration/errors/source-map-support/test.ts:5]: What I mean with this comment is, that the underlying transpiled JS is different than the TS.
So in pure JS this can't happen.

Comment (Jolg42) [reply src/packages/client/src/__tests__/integration/errors/source-map-support/test.ts:5]: New comment is 💯

**Comment by Jolg42** [comment]
New snapshots are 😍
